[CI] RetrySearchIntegTests testSearcherId failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #7801 / opensuse-leap-15_platform-support-unix](https://gradle-enterprise.elastic.co/s/djowjgtre73jy)
- [elasticsearch-periodic-platform-support #7729 / rocky-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/b4lorny3cayyc)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:searchable-snapshots:internalClusterTest" --tests "org.elasticsearch.xpack.searchablesnapshots.RetrySearchIntegTests.testSearcherId" -Dtests.seed=A534D63AB3D7EA48 -Dtests.locale=bn-BD -Dtests.timezone=Asia/Chongqing -Druntime.java=23
```

**Applicable branches:**
8.17

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.searchablesnapshots.RetrySearchIntegTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testSearcherId),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Index{id='jayyxtsbqm/2TCjqpyKSgyYJQyY-VcQvA/DKi0w9foTIO0UUWYFgCi3w/3/_0.cfe/@0', seqNo=2, primaryTerm=2, version=1, autoGeneratedIdTimestamp=-1}
Expected: <4L>
     but: was <2L>
```

**Issue Reasons:**
- [8.17] 2 failures in test testSearcherId (0.5% fail rate in 374 executions)
- [8.17] 2 failures in pipeline elasticsearch-periodic-platform-support (15.4% fail rate in 13 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/RetrySearchIntegTests.java
+++ b/x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/RetrySearchIntegTests.java
@@ -90,6 +90,7 @@ public class RetrySearchIntegTests extends BaseSearchableSnapshotsIntegTestCase 
         for (String allocatedNode : allocatedNodes) {
             if (randomBoolean()) {
                 internalCluster().restartNode(allocatedNode);
+                ensureGreen(indexName);
             }
         }
         ensureGreen(indexName);
@@ -151,6 +152,7 @@ public class RetrySearchIntegTests extends BaseSearchableSnapshotsIntegTestCase 
             final Set<String> allocatedNodes = internalCluster().nodesInclude(indexName);
             for (String allocatedNode : allocatedNodes) {
                 internalCluster().restartNode(allocatedNode);
+                ensureGreen(indexName);
             }
             ensureGreen(indexName);
             assertNoFailuresAndResponse(

